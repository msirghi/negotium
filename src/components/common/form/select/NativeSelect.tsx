import { styled } from '@mui/styles';
import { Select, SelectProps } from '@mui/material';
import colors from '../../../../common/styles/colors';

export const NativeSelect = styled((props: SelectProps<string>) => (
  <Select {...props} native fullWidth variant="outlined" />
))({
  select: {
    '&.MuiOutlinedInput-input': {
      fontSize: 14,
      fontWeight: 400,
      textAlign: 'left',
      padding: '3px 26px 7px 10px',
      backgroundColor: colors.white,
      border: `1px solid ${colors.greys['400']}`,
      transition: 'border-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms',
      '&:focus': {
        borderColor: colors.primaries.lightBlue_1,
      },
      '&.Mui-disabled': {
        WebkitTextFillColor: colors.greys['800'],
        backgroundColor: colors.greys['100'],
      },
    },
  },
  '& .MuiSvgIcon-root': {
    color: colors.primaries.lightBlue_1,
  },
});
