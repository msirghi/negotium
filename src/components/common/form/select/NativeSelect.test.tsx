import renderer from 'react-test-renderer';
import { NativeSelect } from './NativeSelect';

describe('NativeSelect', () => {
  it('should match the snapshot', () => {
    const tree = renderer.create(<NativeSelect />);
    expect(tree).toMatchSnapshot();
  });
});
