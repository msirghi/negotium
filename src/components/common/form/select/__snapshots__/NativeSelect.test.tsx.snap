// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NativeSelect should match the snapshot 1`] = `
<div
  className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiSelect-root Component-root-1 css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
  onClick={[Function]}
>
  <select
    className="MuiNativeSelect-select MuiNativeSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-ffg8md-MuiNativeSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
    multiple={false}
    onAnimationStart={[Function]}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
  />
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiNativeSelect-icon MuiNativeSelect-iconOutlined css-1g12qau-MuiSvgIcon-root-MuiNativeSelect-icon"
    data-testid="ArrowDropDownIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M7 10l5 5 5-5z"
    />
  </svg>
  <fieldset
    aria-hidden={true}
    className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
  >
    <legend
      className="css-1a1ihhj"
    >
      <span
        className="notranslate"
        dangerouslySetInnerHTML={
          Object {
            "__html": "&#8203;",
          }
        }
      />
    </legend>
  </fieldset>
</div>
`;
